spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessagingMessageConverter.java

```diff
@@ -74,6 +74,8 @@ public void setGenerateTimestamp(boolean generateTimestamp) {
 		rawHeaders.put(KafkaHeaders.RECEIVED_TOPIC, record.topic());
 		rawHeaders.put(KafkaHeaders.RECEIVED_PARTITION_ID, record.partition());
 		rawHeaders.put(KafkaHeaders.OFFSET, record.offset());
+		rawHeaders.put(KafkaHeaders.TIMESTAMP_TYPE, record.timestampType().name);
+		rawHeaders.put(KafkaHeaders.TIMESTAMP, record.timestamp());
```

Comment:
M-m-m, would you mind also taking a look into `BatchMessagingMessageConverter` for the same reason to populated these new headers for a batch of records?